File: android/app/src/main/java/com/sdust/iotroniks/data/AppDataManagerHelper.java
package com.sdust.iotroniks.data;

import android.content.Context;
import android.support.annotation.NonNull;

import com.sdust.iotroniks.data.prefs.AppPreferenceHelper;

import java.util.ArrayList;

public class AppDataManagerHelper implements DataManagerHelper {

  private AppPreferenceHelper appPreferenceHelper;

  public AppDataManagerHelper(Context context) {
    this.appPreferenceHelper = new AppPreferenceHelper(context);
  }

  @Override
  public void getIncidents(@NonNull LoadIncidentsCallback loadIncidentsCallback) {
    ArrayList<Incident> incidents = new ArrayList<>();
    loadIncidentsCallback.onIncidentsLoaded(incidents);
  }

  public void setCurrentEmail(String email) {
    appPreferenceHelper.setCurrentUserEmail(email);
  }

  public String getCurrentEmail() {
    return appPreferenceHelper.getCurrentUserEmail();
  }
}

File: android/app/src/main/java/com/sdust/iotroniks/data/AppDataManagerHelper.kt
package com.sdust.iotroniks.data

import android.content.Context
import com.sdust.iotroniks.data.prefs.AppPreferenceHelper
import com.sdust.iotroniks.data.server.AppServerHelper

class AppDataManagerHelper(context: Context) : DataManagerHelper {

  private val appPreferenceHelper: AppPreferenceHelper = AppPreferenceHelper(context)
  private val appServerHelper: AppServerHelper = AppServerHelper()

  init {

  }

  override fun getIncidents(loadIncidentsCallback: DataManagerHelper.LoadIncidentsCallback) {

  }

  override fun getOccurrences(loadOccurrenceCallback: DataManagerHelper.LoadOccurrenceCallback) {
    appServerHelper.getOccurrences(loadOccurrenceCallback)
  }

  override fun setCurrentUserEmail(email: String) {
    TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
  }

  override fun getCurrentUserEmail(): String {
    TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
  }

  var currentEmail: String
    get() = appPreferenceHelper.getCurrentUserEmail()
    set(email) {
      appPreferenceHelper.setCurrentUserEmail(email)
    }
}
